Mathlib/CategoryTheory/Limits/Types/Colimits.lean

```diff
@@ -15,177 +16,100 @@ We show that the category of types has all colimits, by providing the usual conc
-          ext x
-          simp [h]
-        uniq := fun s m hm => by
-          ext x
-          obtain ⟨x, rfl⟩ := e.surjective x
-          obtain ⟨j, x, rfl⟩ := Quot.jointly_surjective x
-          rw [← h, Equiv.apply_symm_apply]
-          simpa [h] using congrFun (hm j) x }
-
-end
+      simpa [f₁, f₂] using ⟨F.ιColimitType j x, by simp⟩
+
+end Functor
+
+namespace Limits.Types
+
+theorem isColimit_iff_coconeTypesIsColimit {F : J ⥤ Type u} (c : Cocone F) :
+    Nonempty (IsColimit c) ↔ (F.coconeTypesEquiv.symm c).IsColimit := by
+  simp only [Functor.CoconeTypes.isColimit_iff, Equiv.apply_symm_apply]
 
 /-- (internal implementation) the colimit cocone of a functor,
 implemented as a quotient of a sigma type
 -/
-@[simps]
-noncomputable def colimitCocone (F : J ⥤ Type u) [Small.{u} (Quot F)] : Cocone F where
-  pt := Shrink (Quot F)
-  ι :=
-    { app := fun j x => equivShrink.{u} _ (Quot.mk _ ⟨j, x⟩)
-      naturality := fun _ _ f => funext fun _ => congrArg _ (Quot.sound ⟨f, rfl⟩).symm }
-
-@[simp]
-theorem Quot.desc_colimitCocone (F : J ⥤ Type u) [Small.{u} (Quot F)] :
-    Quot.desc (colimitCocone F) = equivShrink.{u} (Quot F) := by
-  ext ⟨j, x⟩
-  rfl
+noncomputable abbrev colimitCocone (F : J ⥤ Type u) [Small.{u} F.ColimitType] : Cocone F :=
+  F.coconeTypesEquiv (F.coconeTypes.postcomp (equivShrink.{u} F.ColimitType))
 
 /-- (internal implementation) the fact that the proposed colimit cocone is the colimit -/
-noncomputable def colimitCoconeIsColimit (F : J ⥤ Type u) [Small.{u} (Quot F)] :
+noncomputable def colimitCoconeIsColimit (F : J ⥤ Type u) [Small.{u} F.ColimitType] :
     IsColimit (colimitCocone F) :=
-  Nonempty.some <| by
-    rw [isColimit_iff_bijective_desc, Quot.desc_colimitCocone]
-    exact (equivShrink _).bijective
+  Nonempty.some ((isColimit_iff_coconeTypesIsColimit _).2
+    (F.isColimit_coconeTypes.of_equiv (equivShrink.{u} F.ColimitType) (by aesop)))
+
+theorem hasColimit_iff_small_colimitType (F : J ⥤ Type u) :
+    HasColimit F ↔ Small.{u} F.ColimitType :=
+  ⟨fun _ ↦ small_of_injective
+      ((isColimit_iff_coconeTypesIsColimit _).1 ⟨colimit.isColimit F⟩).bijective.1,
+    fun _ ↦ ⟨_, colimitCoconeIsColimit F⟩⟩
 
-theorem hasColimit_iff_small_quot (F : J ⥤ Type u) : HasColimit F ↔ Small.{u} (Quot F) :=
-  ⟨fun _ => .mk ⟨_, ⟨(Equiv.ofBijective _
-    ((isColimit_iff_bijective_desc (colimit.cocone F)).mp ⟨colimit.isColimit _⟩))⟩⟩,
-   fun _ => ⟨_, colimitCoconeIsColimit F⟩⟩
+@[deprecated (since := "2025-04-01")] alias hasColimit_iff_small_quot :=
+  hasColimit_iff_small_colimitType
 
-theorem small_quot_of_hasColimit (F : J ⥤ Type u) [HasColimit F] : Small.{u} (Quot F) :=
-  (hasColimit_iff_small_quot F).mp inferInstance
+theorem small_colimitType_of_hasColimit (F : J ⥤ Type u) [HasColimit F] :
+    Small.{u} F.ColimitType :=
+  (hasColimit_iff_small_colimitType F).mp inferInstance
+
+@[deprecated (since := "2025-04-01")] alias small_quot_of_hasColimit :=
+  small_colimitType_of_hasColimit
```

Comment:
I can’t comment on that line in github’s interface, but l.150 there is `attribute [local instance] small_quot_of_hasColimit
`, which should be replaced with this. It seems there’s no warning when we `local instance` a deprecated identifier...